src/Memoization.cpp

```diff
@@ -162,6 +163,17 @@ class KeyInfo {
         return (size_t)(1 << i);
     }
 
+// This ifdef is temporary to discuss the tradeoff between the two
+// approaches in code review of the pull request.  Using the full
+// names in the key results in a (hopefulyl incredibly slight)
+// performance differnece based on how one names filters and
+// functions. It is arguably a little easier to debug if something
+// goes wrong as one doesn't need to destructure the cache key. Also,
+// if a pointer is used, a counter must also be put in the cache key
+// to avoid aliasing on reuse of the address in JIT situations where
+// code is regenerated into the same region of memory.
+#define USE_FULL_NAMES_IN_KEY 0
```

Comment:
Obviously we need to decide on this before landing. On my pathological-ish test case it definitely moved the needle performance wise so I vote for setting it to zero.